core/src/main/java/com/linecorp/armeria/server/AbstractBindingBuilder.java

```diff
@@ -0,0 +1,280 @@
+     *
+     * @throws IllegalArgumentException if the specified path pattern is invalid
+     */
+    public AbstractBindingBuilder post(String pathPattern) {
+        addRouteBuilder(pathPattern, EnumSet.of(POST));
+        return this;
+    }
+
+    /**
+     * Sets the path pattern and {@link HttpMethod#PUT} that a {@link Service} will be bound to and support.
+     *
+     * @throws IllegalArgumentException if the specified path pattern is invalid
+     */
+    public AbstractBindingBuilder put(String pathPattern) {
+        addRouteBuilder(pathPattern, EnumSet.of(PUT));
+        return this;
+    }
+
+    /**
+     * Sets the path pattern and {@link HttpMethod#PATCH} that a {@link Service}
+     * will be bound to and support.
+     *
+     * @throws IllegalArgumentException if the specified path pattern is invalid
+     */
+    public AbstractBindingBuilder patch(String pathPattern) {
+        addRouteBuilder(pathPattern, EnumSet.of(PATCH));
+        return this;
+    }
+
+    /**
+     * Sets the path pattern and {@link HttpMethod#DELETE} that a {@link Service}
+     * will be bound to and support.
+     *
+     * @throws IllegalArgumentException if the specified path pattern is invalid
+     */
+    public AbstractBindingBuilder delete(String pathPattern) {
+        addRouteBuilder(pathPattern, EnumSet.of(DELETE));
+        return this;
+    }
+
+    /**
+     * Sets the path pattern and {@link HttpMethod#OPTIONS} that a {@link Service}
+     * will be bound to and support.
+     *
+     * @throws IllegalArgumentException if the specified path pattern is invalid
+     */
+    public AbstractBindingBuilder options(String pathPattern) {
+        addRouteBuilder(pathPattern, EnumSet.of(OPTIONS));
+        return this;
+    }
+
+    /**
+     * Sets the path pattern and {@link HttpMethod#HEAD} that a {@link Service}
+     * will be bound to and support.
+     *
+     * @throws IllegalArgumentException if the specified path pattern is invalid
+     */
+    public AbstractBindingBuilder head(String pathPattern) {
+        addRouteBuilder(pathPattern, EnumSet.of(HEAD));
+        return this;
+    }
+
+    /**
+     * Sets the path pattern and {@link HttpMethod#TRACE} that a {@link Service}
+     * will be bound to and support.
+     *
+     * @throws IllegalArgumentException if the specified path pattern is invalid
+     */
+    public AbstractBindingBuilder trace(String pathPattern) {
+        addRouteBuilder(pathPattern, EnumSet.of(TRACE));
+        return this;
+    }
+
+    /**
+     * Sets the path pattern and {@link HttpMethod#CONNECT} that a {@link Service}
+     * will be bound to and support.
+     *
+     * @throws IllegalArgumentException if the specified path pattern is invalid
+     */
+    public AbstractBindingBuilder connect(String pathPattern) {
+        addRouteBuilder(pathPattern, EnumSet.of(CONNECT));
+        return this;
+    }
+
+    private void addRouteBuilder(String pathPattern, Set<HttpMethod> methods) {
```

Comment:
Actually, it called with two methods.🙃
https://github.com/line/armeria/blob/4d8d6eaff7c9e4fcb77d89daa140725601fc1d14/core/src/main/java/com/linecorp/armeria/server/AbstractBindingBuilder.java#L264
https://github.com/line/armeria/blob/4d8d6eaff7c9e4fcb77d89daa140725601fc1d14/core/src/main/java/com/linecorp/armeria/server/AbstractBindingBuilder.java#L179

Reply:
Doesn't `pathBuilders.forEach(pathBuilder -> addRouteBuilder(pathBuilder, methods)); ` call the below `addRouteBuilder`, not this one?

Reply:
> I think this is only ever called with one method, which would simplify above code a little.

Oops, sorry, This method is called by multiple methods which are `get(pathPattern)`, `post(pathPattern)`...
Did you mean this function is embeded to `get(pathPattern)` and its sibling methods like:
```java
public AbstractBindingBuilder get(String pathPattern) {
    addRouteBuilder(Route.builder().path(requireNonNull(pathPattern, "pathPattern")), EnumSet.of(GET));
    return this;
}

public AbstractBindingBuilder post(String pathPattern) {
    addRouteBuilder(Route.builder().path(requireNonNull(pathPattern, "pathPattern")), EnumSet.of(POST));
    return this;
}
```

Reply:
Ah sorry my suggestion wasn't clear, just that since every caller passes an enumset with one element, I think this could be

```java
private void addRouteBuilder(String pathPattern, HttpMethod method) {
  addRouteBuilder(Route.builder().path(requireNonNull(pathPattern, "pathPattern")), EnumSet.of(method));
}

Reply:
Sorry for the misunderstanding. Thanks for the nice advice. I will address it!